c-api/compat-5.3.c

```diff
@@ -336,22 +392,155 @@ COMPAT53_API void luaL_traceback (lua_State *L, lua_State *L1,
 
 
 COMPAT53_API int luaL_fileresult (lua_State *L, int stat, const char *fname) {
+  const char* s = NULL;
   int en = errno;  /* calls to Lua API may change this value */
   if (stat) {
     lua_pushboolean(L, 1);
     return 1;
   }
   else {
     lua_pushnil(L);
+#if defined(__GLIBC__) || defined(_POSIX_VERSION) || defined(__APPLE__) || (defined(__GNUC__) && (__GNUC__ < 6) && defined (__MINGW32__))
+    // use strerror_r here, because it's available on these specific platforms
+    char buf[512];
+    strerror_r(en, buf, 1024);
```

Comment:
You're passing 1024 as the length, but the buffer is only 512 long.

Reply:
also use the return value, see http://www.club.cc.cmu.edu/~cmccabe/blog_strerror.html

Reply:
Also, why doesn't use strerror_r always, you're defining it above for the platforms that don't have it already, right?

Reply:
`sys_errlist` isn't available everywhere either (I ran into compiler troubles trying to implement it based on that before, too).

Reply:
Well, then you could initialize your buffer with `snprintf(str, str_len, "Unknown error %d", err);`, or at least null-terminate it before your call to strerror_r, if you don't want to handle it differently depending on the GNU or posix implementation.